lib/src/Uncertainty/Distribution/PointConditionalDistribution.cxx

```diff
@@ -222,19 +234,74 @@ class PointConditionalDistributionVBoundEvaluation : public EvaluationImplementa
 };
 
 
+void PointConditionalDistribution::dispatchConditioning(const Collection<Distribution> & distributions, Distribution & simplified) const
+{
+  Collection<Indices> conditioningIndicesBlocks(distributions.getSize());
+  Collection<Point> conditioningValuesBlocks(distributions.getSize());
+  // Sort both the conditioning indices and values in increasing order
+  const UnsignedInteger conditioningSize = conditioningIndices_.getSize();
+  Collection< std::pair<UnsignedInteger, Scalar> > sortedConditioningPairs(conditioningSize);
+  for (UnsignedInteger i = 0; i < conditioningSize; ++i)
+    sortedConditioningPairs[i] = std::pair<UnsignedInteger, Scalar>(conditioningIndices_[i], conditioningValues_[i]);
+  std::sort(sortedConditioningPairs.begin(), sortedConditioningPairs.end());
+  UnsignedInteger beginBlock = 0;
+  UnsignedInteger endBlock = distributions[0].getDimension();
+  UnsignedInteger blockCounter = 0;
+  for (UnsignedInteger conditioningCounter = 0; conditioningCounter < conditioningSize; ++conditioningCounter)
+  {
+    const UnsignedInteger conditioningIndex = sortedConditioningPairs[conditioningCounter].first;
+    // Find the block the conditioning index belongs to
+    while (conditioningIndex >= endBlock)
+    {
+      ++ blockCounter;
+      // Should never go there
+      if (blockCounter == distributions.getSize())
+        break;
+      beginBlock = endBlock;
+      endBlock += distributions[blockCounter].getDimension();
+    }
+    conditioningIndicesBlocks[blockCounter].add(conditioningIndex - beginBlock);
+    conditioningValuesBlocks[blockCounter].add(sortedConditioningPairs[conditioningCounter].second);
+  } // for conditioningCounter
+  // Now, decide what to do for each block
+  Collection<Distribution> newBlocks;
+  for (UnsignedInteger i = 0; i < distributions.getSize(); ++i)
+  {
+    // If the block is fully conditioned, skip it
```

Comment:
You can also code the opposite case: if there is no conditioning, you can directly set the block.